cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/CreateMoneyFlow.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.flow
+
+import co.paralleluniverse.fibers.Suspendable
+import com.example.contract.CashContract
+import com.example.state.CashState
+import net.corda.core.contracts.Command
+import net.corda.core.flows.*
+import net.corda.core.identity.Party
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.transactions.TransactionBuilder
+
+object CreateMoneyFlow {
+    @InitiatingFlow
+    @StartableByRPC
+    class Initiator(private val cashOwners: HashMap<Party, List<Long>>) : FlowLogic<SignedTransaction>() {
+        @Suspendable
+        override fun call(): SignedTransaction {
+            val notary = serviceHub.networkMapCache.notaryIdentities[0]
+
+            val cashCommand = Command(CashContract.Commands.Create(), ourIdentity.owningKey)
+            val txBuilder = TransactionBuilder(notary).addCommand(cashCommand)
+
+            cashOwners.forEach { owner, cashValues ->
+                cashValues.forEach {
+                    val cashState = CashState(it, creator = ourIdentity, owner = owner)
+                    txBuilder.addOutputState(cashState, CashContract.ID)
+                }
+            }
+
+            txBuilder.verify(serviceHub)
+            val signedTx = serviceHub.signInitialTransaction(txBuilder)
+            val setOfSessions: Set<FlowSession> = cashOwners.map { it.key }.map { initiateFlow(it) }.toSet()
```

Comment:
I had this `cashOwners.map { initiateFlow(it.key) }` in previous version of code, but you said, it could potentially create multiple flows to same party, so I can drop `.toSet()` but I think I need map by `key` first, so I don't have duplicates, right?

Reply:
In the previous iteration, IIRC, you had a list with no unicity control. Here you have a map, so there is unicity control on the key, no?